First release of helm chart in new location

torstenwalter opened the pull request:
This is also the first release of the jenkins chart in the new Github org. The README was updated to reflect the new location.

This also adds https://github.com/github/super-linter GitHub action to validate markdown files.
To ensure a nice contributor experience all linting errors have been fixed.

Note: yaml linting is disabled as Helm templates are no valid yaml.

Closes #3

scottrigby commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,3 @@
+# Code of Conduct
```
along with this, delete https://github.com/jenkinsci/helm-charts/blob/main/code-of-conduct.md

torstenwalter replied:
good catch.

scottrigby commented on charts/jenkins/README.md:
````diff
@@ -13,12 +13,19 @@ This chart will do the following:
 
 ```bash
-$ helm install my-release stable/jenkins
+$ helm install my-release jenkinsci/jenkins
````
In general, adopt the README simplified template we added for the new prometheus charts. Example: https://github.com/scottrigby/prometheus-helm-charts/tree/main/charts/prometheus

scottrigby replied:
I made a PR against this PR branch in your fork. https://github.com/torstenwalter/helm-charts/pull/1

timja commented on README.md:
````diff
@@ -0,0 +1,26 @@
+Once Helm is set up properly, add the repo as follows:
+
+```console
+helm repo add jenkinsci https://jenkinsci.github.io/helm-charts
````
we can probably setup a CNAME if that would be preferred?

https://charts.jenkins.io ?

e.g. send a PR here:
https://github.com/jenkins-infra/azure/pull/154/files

torstenwalter replied:
Do you have a name in mind? I've been thinking of `charts.jenkins.io`. Using that the URL would be `helm repo add jenkinsci https://charts.jenkins.io/helm-charts`.

timja replied:
that was the name i had in mine 😄

torstenwalter replied:
Created a PR for it: https://github.com/jenkins-infra/azure/pull/156

torstenwalter replied:
It's been changed to helm repo add jenkinsci https://charts.jenkins.io

oleg-nenashev commented:
CC @jenkinsci/code-reviewers

oleg-nenashev commented on charts/jenkins/CHANGELOG.md:
```diff
@@ -5,6 +5,10 @@ numbering uses [semantic versioning](http://semver.org).
 
 NOTE: The change log until version 1.5.7 is auto generated based on git commits. Those include a reference to the git commit to be able to get more details.
 
+## 2.6.0 First release in jenkinsci GitHub org
```
Should this file get migrated to the repository root? if not, it might make sense to reference it from README

torstenwalter replied:
It\s now referenced from the chart README.

oleg-nenashev commented on charts/jenkins/README.md:
```diff
@@ -79,343 +116,48 @@ So think of the list below more as a general guideline of what should be done.
 
-The following tables list the configurable parameters of the Jenkins chart and their default values.
-
-### Jenkins Master
```
These tables might have had some value as user documentation for those who have no immediate access to Helm, but I'd guess maintaining them is difficult

torstenwalter replied:
@wmcdona89 added them back in a separate file.

oleg-nenashev reviewed (approved):
Looks good overall! Note that the "Jenkins master" terminology is now deprecated, it would be great to cleanup the terminology in subsequent pull requests

scottrigby commented:
Curious why values summary was added back. Generally these tend to get out of sync with values.yaml, which hopefully is already readable-ideally more so.

torstenwalter commented:
https://github.com/torstenwalter/helm-charts/pull/3#discussion_r479716337

torstenwalter commented:
@wmcdona89 I am not really sure if it's a good idea to keep the values table. It's quite some effort to keep it up to date. When removing values or just change the default value of a key one always has to update the table. I think values.yaml should be authorities. We could link to that file in the README instead.

scottrigby commented:
It's really up to you all as maintainers of the chart. In my PR to your branch I just applied the same template as proposed for the prometheus charts, for the same reasons: 1. simplifying (charts have become a bit overcomplicated, the repeated README table convention is just one way), 2. easier maintenance (and support for when users get frustrated when table values don't work as expected, since they're manually copied for each chart.values change and periodically are wrong), and 3. the more clear and simple chart Readmes are overall the better for end users. My opinion though, just wanted to ask if returning those was intentional 😄

torstenwalter commented:
I added some more changes:
- top-level README now at least contains a reference to the Jenkins chart
- migration steps from one version to another is moved into a migration guide section at the bottom. I think it's something people are only interested once, while the more common use case is to look up configuration examples.
- moved the JCasC configuration to the top of the configuration section as this one is an enabler for users as it allows to make the majority of changes.

I also noticed that we still have XML configuration examples in the README. These should be removed or replaces with JCasC alternatives in future PRs.

torstenwalter commented:
I had another look in the file describing the values. What I really like about it is the grouping of related values together. Therefor I am ok with leaving them for now.

Generating it from `values.yaml` will not provide us the same experience. Using new keys for values to do the grouping will be a breaking change. What we could try is to sort them in the `value.yaml` the same way and use comments for seperation. Downside is that keys are not sorted alphabetically.

torstenwalter commented:
So this PR is up for a last review round. We should try to move future improvements to new PRs to unblock the migration and allow people to contribute.

wmcdona89 commented:
I opened issue https://github.com/jenkinsci/helm-charts/issues/18 to organize `values.yaml` to align with the sections in `VALUES_SUMMARY`. then the `README` can link to each section and `VALUES_SUMMARY` can be removed.
